### PR TITLE
enabled optional chaining code parsing and fixing

### PR DESCRIPTION
**Pull request by SalvatorePreviti**
- modify the expression interpolator so we can pass constant strings too to the expression
- change our current replacers to support typescript and javascript optional chaining using ['', '?', '!'].map
- fix the existing code to handle optional call expression
- fix the existing code to handle the optional member expression
- add in the tests enough ?. and !. to test the support of optional chaining and not null expressions

### CODE REVIEW

File: packages/codemods/src/transforms/transform-grid-api-methods-v31/replacements.ts [hunk 1]
```diff
@@ -7,106 +7,141 @@ import {
...
 const GRID_API_DEPRECATIONS: Array<GridApiDeprecation> = [
-  matchNode(({ api }) => ast.expression`${api}.setGetRowId()`, {
-    api: p.expression(),
-  }),
-];
-
-const COLUMNS_API_REPLACEMENTS: Array<GridApiReplacement> = [
-  replace(
-    matchNode(({ api }) => ast.expression`${api}.getAllColumns()`, {
+  ...['', '?', '!'].map((apiCoalesce) =>
```
Comment (StephenCooper) [inline packages/codemods/src/transforms/transform-grid-api-methods-v31/replacements.ts:84]: Do you think we should push this up and change the format of the config to return a function(apiCoalesce) => array? Save repeating that in every generator?

**Comment by StephenCooper** [comment]
There are a number of templates that are used by the code scaffolding scripts that will also need updating once we have decided if this is the correct approach. That way when they are used at least the examples will be correct to copy from.

File: packages/codemods/src/transforms/transform-grid-api-methods-v31-3/__fixtures__/scenarios/hello-world/input.js [hunk 2]
```diff
@@ -8,4 +8,4 @@ const gridApi = createGrid(document.body, {
 const col = 'foo';
 const row = {};
 gridApi.getValue(col, row);
-gridApi.getValue('colId', row);
\ No newline at end of file
+gridApi?.getValue('colId', row);
```
Comment (StephenCooper) [inline packages/codemods/src/transforms/transform-grid-api-methods-v31-3/__fixtures__/scenarios/hello-world/input.js:11]: We should cover each framework with a test case to make sure it works as expected. i.e vue, react, angular as well as the ts and js tests you have here.

Comment (SalvatorePreviti) [reply packages/codemods/src/transforms/transform-grid-api-methods-v31-3/__fixtures__/scenarios/hello-world/input.js:11]: yes, cleanup and tests are something I am going to work on next

File: packages/codemods/src/transforms/transform-grid-api-methods-v31/replacements.ts (hunk 1 above)
Comment (SalvatorePreviti) [reply packages/codemods/src/transforms/transform-grid-api-methods-v31/replacements.ts:84]: Well yes I thought about it but looked much more complex than this - i will give it another try.
There is also the problem that we might run multiple times transformations that don't use the apiCoalesce (that I will rename by the way)

Comment (StephenCooper) [reply packages/codemods/src/transforms/transform-grid-api-methods-v31/replacements.ts:84]: Ok if this makes it more complex maybe lets not worry about that.

Comment (SalvatorePreviti) [reply packages/codemods/src/transforms/transform-grid-api-methods-v31/replacements.ts:84]: We agreed that the number of changes required to make this work would be too big - for now we can live with this and consider a bigger refactoring in the future. The advantage of this is, whilst is more verbose, is very clear what is happening.

**Pull request merged**